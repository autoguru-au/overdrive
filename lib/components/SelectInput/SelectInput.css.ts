import { style } from '@vanilla-extract/css';

import { overdriveTokens as vars } from '../../themes/theme.css';

export const input = style({
	marginBottom: `calc(-1 * ${vars.border.width['1']})`,
	marginTop: `calc(-1 * ${vars.border.width['1']})`,
});
export const paddedInput = style({
	selectors: {
		[`${input}&`]: {
			paddingRight: vars.space['8'],
		},
	},
});

export const arrow = style({
	right: '0',
	top: '0',
});

export const arrowDisabled = style({
	opacity: 0.2,
});
