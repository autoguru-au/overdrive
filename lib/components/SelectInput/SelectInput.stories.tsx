import { CarIcon } from '@autoguru/icons';
import { action } from '@storybook/addon-actions';
import type { Meta, StoryObj } from '@storybook/react';
import React, { type ComponentProps } from 'react';

import { argTypesExampleIcons } from '../../stories/shared/argTypes';
import { DateInput } from '../DateInput';

import { SelectInput } from '.';

const valueOptions = ['Kia', 'Toyota', 'BMW', 'Mitsubishi', 'Hyundai'];
const selectOptions = valueOptions.map((item) => (
	<option key={item} value={item}>
		{item}
	</option>
));

const defaultValue = valueOptions[4];
const defaultPlaceholder = 'What is the make of your car?';

const attachOptions: Record<
	string,
	ComponentProps<typeof DateInput>['attach']
> = {
	NONE: 'NONE',
	TOP: 'TOP',
	RIGHT: 'RIGHT',
	LEFT: 'LEFT',
	BOTTOM: 'BOTTOM',
	ALL: 'ALL',
};

const meta: Meta<typeof SelectInput> = {
	title: 'Forms & Input Fields/Select',
	component: SelectInput,
	args: {
		// @ts-expect-error null is assignable to type 'string | undefined'
		value: null,
		disabled: false,
		name: 'text',
		placeholder: defaultPlaceholder,
		isValid: false,
		isTouched: false,
		isLoading: false,
		isFocused: false,
		reserveHintSpace: false,
		hintText: '',
		notch: true,
		attach: 'NONE',
		prefixIcon: CarIcon,
		onChange: action('onChange'),
		onFocus: action('onFocus'),
		onBlur: action('onBlur'),
		children: (
			<>
				<option disabled>Select an option</option>
				{selectOptions}
			</>
		),
	},
	argTypes: {
		value: {
			options: valueOptions,
			control: {
				type: 'select',
			},
		},
		attach: {
			description: 'Input attach',
			options: Object.keys(attachOptions),
			control: {
				type: 'select',
			},
		},
		prefixIcon: {
			...argTypesExampleIcons,
			description: 'Input prefix Icon',
		},
	},
};

export default meta;
type Story = StoryObj<typeof meta>;

/**
 * Additional examples of shared input field states and variants can be seen in
 * [Text Input](/docs/forms-input-fields-text-input--docs)
 */
export const Standard: Story = {
	args: {
		hintText: 'Hint Text',
	},
};

export const Valid: Story = {
	args: {
		value: defaultValue,
		placeholder: defaultPlaceholder,
		isTouched: true,
		isValid: true,
	},
};

export const Invalid: Story = {
	args: {
		placeholder: defaultPlaceholder,
		isTouched: true,
		isValid: false,
		hintText: 'Vehicle make is mandatory',
	},
};

export const Disabled: Story = {
	args: {
		value: defaultValue,
		placeholder: defaultPlaceholder,
		disabled: true,
	},
};

export const SizeSmall: Story = {
	args: {
		size: 'small',
	},
};
