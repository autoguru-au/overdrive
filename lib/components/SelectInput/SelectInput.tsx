import React, { memo } from 'react';
import { ChevronDownIcon, Icon } from '../Icon';
import { withEnhancedInput } from '../InputBase';
import styles from './style.scss';
import clsx from 'clsx';

const SelectInputComponent = ({
	field,
	eventHandlers,
	suffixed,
	prefixed,
	validation,
	...rest
}) => (
	<>
		<select {...eventHandlers} {...field} {...rest} autoComplete="off" />
		<Icon
			className={clsx(styles.arrow, { [styles.suffixed]: suffixed })}
			size={25}
			icon={ChevronDownIcon}
		/>
	</>
);

SelectInputComponent.primitiveType = 'select';

export const SelectInput = memo(
	withEnhancedInput(SelectInputComponent, {
		withSuffixIcon: false,
		withForcedSuffixIconPadding: true,
	}),
);
