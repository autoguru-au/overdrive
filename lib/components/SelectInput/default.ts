export { SelectInput as default } from './SelectInput';
