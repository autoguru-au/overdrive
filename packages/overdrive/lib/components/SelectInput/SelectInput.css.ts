import { style } from '@vanilla-extract/css';

import { themeContractVars as vars } from '../../themes/theme.css';

export const input = style({});
export const paddedInput = style({
	selectors: {
		[`${input}&`]: {
			paddingRight: vars.space['8'],
		},
	},
});

export const arrow = style({
	top: '0',
	right: '0',
});
