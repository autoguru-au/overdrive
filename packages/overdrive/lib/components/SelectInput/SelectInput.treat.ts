import { style } from 'treat';

export const input = style({});
export const paddedInput = style(({ space }) => ({
	selectors: {
		[`${input}&`]: {
			paddingRight: space['8'],
		},
	},
}));

export const arrow = style({
	top: '0',
	right: '0',
});
