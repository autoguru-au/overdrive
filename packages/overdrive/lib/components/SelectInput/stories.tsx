import { AccountEditIcon, CalendarIcon } from '@autoguru/icons';
import { boolean, text } from '@storybook/addon-knobs';
import * as React from 'react';

import { SelectInput } from '.';

const isTouched = (touched: boolean) => boolean('isTouched', touched);
const isValid = (valid: boolean) => boolean('isValid', valid);

const selectOptions = (optionsCSV) =>
	text('Options', optionsCSV)
		.split(',')
		.map((item) => item.trim())
		.map((item) => (
			<option key={item} value={item}>
				{item}
			</option>
		));

export default {
	title: 'Components/Inputs/Select',
	component: SelectInput,
	parameters: {
		chromatic: { delay: 300 },
	},
};

export const standard = () => (
	<SelectInput name="abc" placeholder="Select one">
		<option disabled />
		{selectOptions('Option 1, Option 2, Option 3')}
	</SelectInput>
);
export const withAValue = () => (
	<SelectInput name="abc" placeholder="Select one" value="Option 2">
		<option disabled />
		{selectOptions('Option 1, Option 2, Option 3')}
	</SelectInput>
);
export const withHintText = () => (
	<SelectInput
		name="abc"
		placeholder="Select one"
		hintText={text('Hint Text', 'Cannot select option 3')}>
		<option disabled />
		{selectOptions('Option 1, Option 2, Option 3')}
	</SelectInput>
);
export const withIcon = () => (
	<div style={{ display: 'grid', gridGap: '16px' }}>
		<SelectInput
			name="abc"
			placeholder="Select one"
			prefixIcon={CalendarIcon}>
			<option disabled />
			{selectOptions('Option 1, Option 2, Option 3')}
		</SelectInput>

		<SelectInput
			isTouched
			isValid
			name="abc"
			placeholder="Select one"
			prefixIcon={AccountEditIcon}>
			<option disabled />
			{selectOptions('Option 1, Option 2, Option 3')}
		</SelectInput>
		<SelectInput
			isTouched
			name="abc"
			placeholder="Select one"
			prefixIcon={AccountEditIcon}
			isValid={false}>
			<option disabled />
			{selectOptions('Option 1, Option 2, Option 3')}
		</SelectInput>
		<SelectInput
			isTouched
			disabled
			name="abc"
			placeholder="Select one"
			prefixIcon={AccountEditIcon}
			isValid={false}>
			<option disabled />
			{selectOptions('Option 1, Option 2, Option 3')}
		</SelectInput>
	</div>
);
export const withValidation = () => (
	<SelectInput
		name="abc"
		isValid={isValid(false)}
		isTouched={isTouched(false)}
		placeholder="Select one"
		hintText={text('Hint Text', 'Cannot select option 3')}>
		<option disabled />
		{selectOptions('Option 1, Option 2, Option 3')}
	</SelectInput>
);
export const disabled = () => (
	<SelectInput disabled name="abc" placeholder="Select one" value="Option 2">
		<option disabled />
		<option value="Option 2">Option 2</option>
	</SelectInput>
);

export const loading = () => (
	<SelectInput isLoading name="abc" placeholder="Select one" value="Option 2">
		<option disabled />
		<option value="Option 2">Option 2</option>
	</SelectInput>
);

export const loadingWithIcon = () => (
	<SelectInput
		isLoading
		prefixIcon={AccountEditIcon}
		name="abc"
		placeholder="Select one"
		value="Option 2">
		<option disabled />
		<option value="Option 2">Option 2</option>
	</SelectInput>
);
