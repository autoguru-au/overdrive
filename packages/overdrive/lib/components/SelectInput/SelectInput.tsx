import { ChevronDownIcon } from '@autoguru/icons';
import * as React from 'react';
import { ReactNode } from 'react';
import { useStyles } from 'react-treat';

import { Box } from '../Box';
import { Icon } from '../Icon';
import { withEnhancedInput } from '../private/InputBase';

import * as styleRefs from './SelectInput.treat';

export const SelectInput = withEnhancedInput<
	{
		children: ReactNode[];
	},
	HTMLSelectElement
>(
	({
		field,
		eventHandlers,
		suffixed,
		prefixed,
		validation,
		isLoading,
		...rest
	}) => {
		const styles = useStyles(styleRefs);
		return (
			<Box
				display="flex"
				flexWrap="nowrap"
				alignItems="center"
				justifyContent="center"
				position="relative">
				<Box
					is="select"
					flexGrow={1}
					{...eventHandlers}
					{...field}
					{...rest}
					className={[
						styles.input,
						styles.paddedInput,
						field.className,
					]}
					autoComplete="off"
				/>
				{isLoading ? null : (
					<Box
						className={styles.arrow}
						display="flex"
						alignItems="center"
						height="full"
						marginRight="4"
						flexShrink={0}
						pointerEvents="none"
						position="absolute">
						<Icon size="medium" icon={ChevronDownIcon} />
					</Box>
				)}
			</Box>
		);
	},
	{
		primitiveType: 'select',
		withSuffixIcon: false,
	},
);
