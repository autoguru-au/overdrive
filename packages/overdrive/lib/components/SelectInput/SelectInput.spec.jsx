import { fireEvent, render } from '@testing-library/react';
import * as React from 'react';

import { SelectInput } from './SelectInput';

const TestIcon = (
	<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">
		<path d="M183.253 353.707L280.96 256l-97.707-97.92 30.08-30.08 128 128-128 128-30.08-30.293z" />
	</svg>
);

// TODO: Confirm these tests are actually valid and working? ️¯\_(ツ)_/¯
describe('<SelectInput />', () => {
	it('should match snapshot', () => {
		expect(
			render(
				<SelectInput placeholder="Hello World!">
					<option value="a">Value 1</option>
					<option value="b">Value 2</option>
					<option value="c">Value 3</option>
				</SelectInput>,
			).container.firstChild,
		).toMatchSnapshot();
	});

	it('should have some hintText', () => {
		const hintText = () => 'hint text';

		const { container } = render(
			<SelectInput placeholder="Hello World!" hintText={hintText()}>
				<option value="a">Value 1</option>
				<option value="b">Value 2</option>
				<option value="c">Value 3</option>
			</SelectInput>,
		);

		expect(container.firstChild).toMatchSnapshot();

		expect(container).toHaveTextContent(hintText());
	});

	it('should have an select element', () => {
		const { container } = render(
			<SelectInput placeholder="Hello World!">
				<option value="a">Value 1</option>
				<option value="b">Value 2</option>
				<option value="c">Value 3</option>
			</SelectInput>,
		);
		expect(container.querySelector('select')).toBeInTheDocument();
	});

	it.skip('should pass on className to dom element', () => {
		const { container } = render(
			<SelectInput placeholder="Hello World!" className="input-class">
				<option value="a">Value 1</option>
				<option value="b">Value 2</option>
				<option value="c">Value 3</option>
			</SelectInput>,
		);
		expect(container.firstChild).toHaveClass('input-class');
	});

	it('should match snapshot when active', () => {
		const { container } = render(
			<SelectInput placeholder="Hello World!">
				<option value="a">Value 1</option>
				<option value="b">Value 2</option>
				<option value="c">Value 3</option>
			</SelectInput>,
		);

		fireEvent.focus(container.querySelector('select'));

		fireEvent.change(container.querySelector('select'), {
			target: { value: 'a' },
		});

		expect(container.firstChild).toMatchSnapshot();
	});

	it('should match snapshot when touched', () => {
		expect(
			render(
				<SelectInput isTouched placeholder="Hello World!">
					<option value="a">Value 1</option>
					<option value="b">Value 2</option>
					<option value="c">Value 3</option>
				</SelectInput>,
			).container.firstChild,
		).toMatchSnapshot();
	});

	it('should match snapshot when touched and valid', () => {
		expect(
			render(
				<SelectInput isTouched isValid placeholder="Hello World!">
					<option value="a">Value 1</option>
					<option value="b">Value 2</option>
					<option value="c">Value 3</option>
				</SelectInput>,
			).container.firstChild,
		).toMatchSnapshot();
	});

	it('should match snapshot when touched and invalid', () => {
		expect(
			render(
				<SelectInput
					isTouched
					placeholder="Hello World!"
					isValid={false}>
					<option value="a">Value 1</option>
					<option value="b">Value 2</option>
					<option value="c">Value 3</option>
				</SelectInput>,
			).container.firstChild,
		).toMatchSnapshot();
	});

	it('should match snapshot with prefix icon', () => {
		expect(
			render(
				<SelectInput
					className="input-class"
					placeholder="placeholder something"
					prefixIcon={TestIcon}
				/>,
			).container.firstChild,
		).toMatchSnapshot();
	});

	it('should match snapshot with custom icon', () => {
		expect(
			render(
				<SelectInput
					className="input-class"
					placeholder="placeholder something"
					fieldIcon={TestIcon}
				/>,
			).container.firstChild,
		).toMatchSnapshot();
	});

	it('should display placeholder text', () => {
		const { container } = render(
			<SelectInput placeholder="placeholder something">
				<option value="a">Value 1</option>
				<option value="b">Value 2</option>
				<option value="c">Value 3</option>
			</SelectInput>,
		);

		expect(container.querySelector('label')).toHaveTextContent(
			'placeholder something',
		);
	});

	it('should fire onFocus event', () => {
		const spyedCallback = jest.fn();

		const { container } = render(
			<SelectInput placeholder="Hello World!" onFocus={spyedCallback}>
				<option value="a">Value 1</option>
				<option value="b">Value 2</option>
				<option value="c">Value 3</option>
			</SelectInput>,
		);

		fireEvent.focus(container.querySelector('select'));

		expect(spyedCallback).toHaveBeenCalledTimes(1);
	});

	it('should fire onBlur event', () => {
		const spyedCallback = jest.fn();

		const { container } = render(
			<SelectInput placeholder="Hello World!" onBlur={spyedCallback}>
				<option value="a">Value 1</option>
				<option value="b">Value 2</option>
				<option value="c">Value 3</option>
			</SelectInput>,
		);

		fireEvent.blur(container.querySelector('select'));

		expect(spyedCallback).toHaveBeenCalledTimes(1);
	});

	it('should fire onChange event', () => {
		const spyedCallback = jest.fn();

		const { container } = render(
			<SelectInput placeholder="Hello World!" onChange={spyedCallback}>
				<option value="a">Value 1</option>
				<option value="b">Value 2</option>
				<option value="c">Value 3</option>
			</SelectInput>,
		);

		fireEvent.change(container.querySelector('select'), {
			target: { value: 'test' },
		});

		expect(spyedCallback).toHaveBeenCalledTimes(1);
	});
});
