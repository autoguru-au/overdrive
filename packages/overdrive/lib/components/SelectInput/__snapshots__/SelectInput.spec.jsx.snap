// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectInput /> should have some hintText 1`] = `
<div
  class="base width_full"
>
  <div
    class="base none_mobile_base none_mobile_base none_mobile_base none_mobile_base width_full position_relative root_base"
  >
    <div
      class="base width_full height_full"
    >
      <div
        class="base display_flex position_relative center_mobile_base flexWrap_nowrap center_mobile_base"
      >
        <select
          autocomplete="off"
          class="base block appearance select flexGrow_1 input paddedInput_base base block appearance input display_flex width_full position_relative input_itself_root_base"
        >
          <option
            value="a"
          >
            Value 1
          </option>
          <option
            value="b"
          >
            Value 2
          </option>
          <option
            value="c"
          >
            Value 3
          </option>
        </select>
        <div
          class="base 4_mobile_base display_flex height_full position_absolute pointerEvents_none center_mobile_base flexShrink_0 arrow"
        >
          <i
            class="base display_block medium_mobile_base"
            role="presentation"
          >
            <svg
              class="base display_block width_full height_full"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.413 8.584L12 13.17l4.586-4.586L18 9.998l-6 6-6-6 1.413-1.414z"
              />
            </svg>
          </i>
        </div>
      </div>
    </div>
    <div
      class="base display_flex width_full height_full position_absolute textAlign_left pointerEvents_none borderVisualDefaults_base borders_root_default"
    >
      <div
        class="borderRegionDefaults_base borders_leading_base styleNode_base"
      />
      <div
        class="borderRegionDefaults_base borders_middle_base styleNode_base"
        style="width: 0px;"
      >
        <label
          class="base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base display_inline position_absolute pointerEvents_none placeholder_default_base root_base colours_neutral_base noWrap sizes_4_base placeholder_mutedLabelStyles_base placeholderPlacement_default_base"
        >
          Hello World!
        </label>
      </div>
      <div
        class="borderRegionDefaults_base borders_trailing_base styleNode_base"
      />
    </div>
  </div>
  <p
    class="base textAlign_left root_base fontWeight_normal_base sizes_2_base 2_mobile_base 4_mobile_base hintText_base styleNode_base"
  >
    hint text
  </p>
</div>
`;

exports[`<SelectInput /> should match snapshot 1`] = `
<div
  class="base width_full"
>
  <div
    class="base none_mobile_base none_mobile_base none_mobile_base none_mobile_base width_full position_relative root_base"
  >
    <div
      class="base width_full height_full"
    >
      <div
        class="base display_flex position_relative center_mobile_base flexWrap_nowrap center_mobile_base"
      >
        <select
          autocomplete="off"
          class="base block appearance select flexGrow_1 input paddedInput_base base block appearance input display_flex width_full position_relative input_itself_root_base"
        >
          <option
            value="a"
          >
            Value 1
          </option>
          <option
            value="b"
          >
            Value 2
          </option>
          <option
            value="c"
          >
            Value 3
          </option>
        </select>
        <div
          class="base 4_mobile_base display_flex height_full position_absolute pointerEvents_none center_mobile_base flexShrink_0 arrow"
        >
          <i
            class="base display_block medium_mobile_base"
            role="presentation"
          >
            <svg
              class="base display_block width_full height_full"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.413 8.584L12 13.17l4.586-4.586L18 9.998l-6 6-6-6 1.413-1.414z"
              />
            </svg>
          </i>
        </div>
      </div>
    </div>
    <div
      class="base display_flex width_full height_full position_absolute textAlign_left pointerEvents_none borderVisualDefaults_base borders_root_default"
    >
      <div
        class="borderRegionDefaults_base borders_leading_base styleNode_base"
      />
      <div
        class="borderRegionDefaults_base borders_middle_base styleNode_base"
        style="width: 0px;"
      >
        <label
          class="base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base display_inline position_absolute pointerEvents_none placeholder_default_base root_base colours_neutral_base noWrap sizes_4_base placeholder_mutedLabelStyles_base placeholderPlacement_default_base"
        >
          Hello World!
        </label>
      </div>
      <div
        class="borderRegionDefaults_base borders_trailing_base styleNode_base"
      />
    </div>
  </div>
</div>
`;

exports[`<SelectInput /> should match snapshot when active 1`] = `
<div
  class="base width_full"
>
  <div
    class="base none_mobile_base none_mobile_base none_mobile_base none_mobile_base width_full position_relative root_base"
  >
    <div
      class="base width_full height_full"
    >
      <div
        class="base display_flex position_relative center_mobile_base flexWrap_nowrap center_mobile_base"
      >
        <select
          autocomplete="off"
          class="base block appearance select flexGrow_1 input paddedInput_base base block appearance input display_flex width_full position_relative input_itself_root_base"
        >
          <option
            value="a"
          >
            Value 1
          </option>
          <option
            value="b"
          >
            Value 2
          </option>
          <option
            value="c"
          >
            Value 3
          </option>
        </select>
        <div
          class="base 4_mobile_base display_flex height_full position_absolute pointerEvents_none center_mobile_base flexShrink_0 arrow"
        >
          <i
            class="base display_block medium_mobile_base"
            role="presentation"
          >
            <svg
              class="base display_block width_full height_full"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.413 8.584L12 13.17l4.586-4.586L18 9.998l-6 6-6-6 1.413-1.414z"
              />
            </svg>
          </i>
        </div>
      </div>
    </div>
    <div
      class="base display_flex width_full height_full position_absolute textAlign_left pointerEvents_none borderVisualDefaults_base borders_root_default"
    >
      <div
        class="borderRegionDefaults_base borders_leading_base activeBorderColour_base"
      />
      <div
        class="borderRegionDefaults_base borders_middle_base activeBorderColour_base"
        style="width: 16px;"
      >
        <label
          class="base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base display_inline position_absolute pointerEvents_none placeholder_default_base activeColour_base root_base colours_neutral_base noWrap sizes_4_base placeholderPlacement_shifted_base"
        >
          Hello World!
        </label>
      </div>
      <div
        class="borderRegionDefaults_base borders_trailing_base activeBorderColour_base"
      />
    </div>
  </div>
</div>
`;

exports[`<SelectInput /> should match snapshot when touched 1`] = `
<div
  class="base width_full"
>
  <div
    class="base none_mobile_base none_mobile_base none_mobile_base none_mobile_base width_full position_relative root_base"
  >
    <div
      class="base width_full height_full"
    >
      <div
        class="base display_flex position_relative center_mobile_base flexWrap_nowrap center_mobile_base"
      >
        <select
          autocomplete="off"
          class="base block appearance select flexGrow_1 input paddedInput_base base block appearance input display_flex width_full position_relative input_itself_root_base"
        >
          <option
            value="a"
          >
            Value 1
          </option>
          <option
            value="b"
          >
            Value 2
          </option>
          <option
            value="c"
          >
            Value 3
          </option>
        </select>
        <div
          class="base 4_mobile_base display_flex height_full position_absolute pointerEvents_none center_mobile_base flexShrink_0 arrow"
        >
          <i
            class="base display_block medium_mobile_base"
            role="presentation"
          >
            <svg
              class="base display_block width_full height_full"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.413 8.584L12 13.17l4.586-4.586L18 9.998l-6 6-6-6 1.413-1.414z"
              />
            </svg>
          </i>
        </div>
      </div>
    </div>
    <div
      class="base display_flex width_full height_full position_absolute textAlign_left pointerEvents_none borderVisualDefaults_base borders_root_default"
    >
      <div
        class="borderRegionDefaults_base borders_leading_base styleNode_base"
      />
      <div
        class="borderRegionDefaults_base borders_middle_base styleNode_base"
        style="width: 0px;"
      >
        <label
          class="base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base display_inline position_absolute pointerEvents_none placeholder_default_base root_base colours_neutral_base noWrap sizes_4_base placeholder_mutedLabelStyles_base placeholderPlacement_default_base"
        >
          Hello World!
        </label>
      </div>
      <div
        class="borderRegionDefaults_base borders_trailing_base styleNode_base"
      />
    </div>
  </div>
</div>
`;

exports[`<SelectInput /> should match snapshot when touched and invalid 1`] = `
<div
  class="base width_full"
>
  <div
    class="base none_mobile_base none_mobile_base none_mobile_base none_mobile_base width_full position_relative root_base"
  >
    <div
      class="base width_full height_full"
    >
      <div
        class="base display_flex position_relative center_mobile_base flexWrap_nowrap center_mobile_base"
      >
        <select
          autocomplete="off"
          class="base block appearance select flexGrow_1 input paddedInput_base base block appearance input display_flex width_full position_relative input_itself_root_base"
        >
          <option
            value="a"
          >
            Value 1
          </option>
          <option
            value="b"
          >
            Value 2
          </option>
          <option
            value="c"
          >
            Value 3
          </option>
        </select>
        <div
          class="base 4_mobile_base display_flex height_full position_absolute pointerEvents_none center_mobile_base flexShrink_0 arrow"
        >
          <i
            class="base display_block medium_mobile_base"
            role="presentation"
          >
            <svg
              class="base display_block width_full height_full"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.413 8.584L12 13.17l4.586-4.586L18 9.998l-6 6-6-6 1.413-1.414z"
              />
            </svg>
          </i>
        </div>
      </div>
    </div>
    <div
      class="base display_flex width_full height_full position_absolute textAlign_left pointerEvents_none borderVisualDefaults_base borders_root_default"
    >
      <div
        class="borderRegionDefaults_base borders_leading_base styleNode_base"
      />
      <div
        class="borderRegionDefaults_base borders_middle_base styleNode_base"
        style="width: 0px;"
      >
        <label
          class="base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base display_inline position_absolute pointerEvents_none placeholder_default_base root_base colours_neutral_base noWrap sizes_4_base placeholder_mutedLabelStyles_base placeholderPlacement_default_base"
        >
          Hello World!
        </label>
      </div>
      <div
        class="borderRegionDefaults_base borders_trailing_base styleNode_base"
      />
    </div>
  </div>
</div>
`;

exports[`<SelectInput /> should match snapshot when touched and valid 1`] = `
<div
  class="base width_full"
>
  <div
    class="base none_mobile_base none_mobile_base none_mobile_base none_mobile_base width_full position_relative root_base"
  >
    <div
      class="base width_full height_full"
    >
      <div
        class="base display_flex position_relative center_mobile_base flexWrap_nowrap center_mobile_base"
      >
        <select
          autocomplete="off"
          class="base block appearance select flexGrow_1 input paddedInput_base base block appearance input display_flex width_full position_relative input_itself_root_base"
        >
          <option
            value="a"
          >
            Value 1
          </option>
          <option
            value="b"
          >
            Value 2
          </option>
          <option
            value="c"
          >
            Value 3
          </option>
        </select>
        <div
          class="base 4_mobile_base display_flex height_full position_absolute pointerEvents_none center_mobile_base flexShrink_0 arrow"
        >
          <i
            class="base display_block medium_mobile_base"
            role="presentation"
          >
            <svg
              class="base display_block width_full height_full"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.413 8.584L12 13.17l4.586-4.586L18 9.998l-6 6-6-6 1.413-1.414z"
              />
            </svg>
          </i>
        </div>
      </div>
    </div>
    <div
      class="base display_flex width_full height_full position_absolute textAlign_left pointerEvents_none borderVisualDefaults_base borders_root_default"
    >
      <div
        class="borderRegionDefaults_base borders_leading_base styleNode_base"
      />
      <div
        class="borderRegionDefaults_base borders_middle_base styleNode_base"
        style="width: 0px;"
      >
        <label
          class="base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base display_inline position_absolute pointerEvents_none placeholder_default_base root_base colours_neutral_base noWrap sizes_4_base placeholder_mutedLabelStyles_base placeholderPlacement_default_base"
        >
          Hello World!
        </label>
      </div>
      <div
        class="borderRegionDefaults_base borders_trailing_base styleNode_base"
      />
    </div>
  </div>
</div>
`;

exports[`<SelectInput /> should match snapshot with prefix icon 1`] = `
<div
  class="base width_full input-class"
>
  <div
    class="base none_mobile_base none_mobile_base none_mobile_base none_mobile_base width_full position_relative root_base"
  >
    <div
      class="base width_full height_full"
    >
      <i
        class="base display_block medium_mobile_base position_absolute pointerEvents_none iconRoot_base icon_prefix styleNode_base"
        role="presentation"
      >
        <svg
          class="base display_block width_full height_full"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M183.253 353.707L280.96 256l-97.707-97.92 30.08-30.08 128 128-128 128-30.08-30.293z"
          />
        </svg>
      </i>
      <div
        class="base display_flex position_relative center_mobile_base flexWrap_nowrap center_mobile_base"
      >
        <select
          autocomplete="off"
          class="base block appearance select flexGrow_1 input paddedInput_base base block appearance input display_flex width_full position_relative input_itself_root_base input_itself_prefixed_base"
        />
        <div
          class="base 4_mobile_base display_flex height_full position_absolute pointerEvents_none center_mobile_base flexShrink_0 arrow"
        >
          <i
            class="base display_block medium_mobile_base"
            role="presentation"
          >
            <svg
              class="base display_block width_full height_full"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.413 8.584L12 13.17l4.586-4.586L18 9.998l-6 6-6-6 1.413-1.414z"
              />
            </svg>
          </i>
        </div>
      </div>
    </div>
    <div
      class="base display_flex width_full height_full position_absolute textAlign_left pointerEvents_none borderVisualDefaults_base borders_root_default"
    >
      <div
        class="borderRegionDefaults_base borders_leading_base styleNode_base"
      />
      <div
        class="borderRegionDefaults_base borders_middle_base styleNode_base"
        style="width: 0px;"
      >
        <label
          class="base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base none_mobile_base display_inline position_absolute pointerEvents_none placeholder_default_base root_base colours_neutral_base noWrap sizes_4_base placeholder_mutedLabelStyles_base placeholderPlacement_defaultPrefixed_base"
        >
          placeholder something
        </label>
      </div>
      <div
        class="borderRegionDefaults_base borders_trailing_base styleNode_base"
      />
    </div>
  </div>
</div>
`;
